pythia/models/butd.py

```diff
@@ -119,41 +119,36 @@ def forward(self, sample_list):
             dtype=torch.float,
         )
 
-        # For beam search inference. Currently beam seach for BUTD works only 
-        # with batch_size = 1 and should be used with run_type inference only.
-        # TODO : Implement batch beam search
-        if self.config["inference"]["type"] == "beam_search":
-            beam_search = BeamSearch(
+        if self.config["inference"]["type"] == "greedy":
+            decoder = BeamSearch(self.vocab, 1)
```

Comment:
Can you pass self.vocab, self.config to the BeamSearch() and and NucleusSampling functions and extract the useful value inside the init function instead of the if condition here?

Reply:
Yeah about that. It would still require a conditional statement in Beam Search's init function because we want to use Greedy using Beam Search object only. Also, with that approach, how do we make a call for Greedy scenario without using if?

Reply:
Amm, can we not set the beam_length in the config to 1 for greedy decoding. This way, the design will be clean and we will be able to get rid of this if statement.

Reply:
Yep. This should be doable, as long as we're okay with having mentions of 'beam' in greedy's config.

Reply:
I've tried to simplify it. Let me know if the latest version works?